src/components/Coldcard/ColdcardSigner.jsx

```diff
@@ -40,28 +33,36 @@ class ColdcardSigner extends Component {
       ? "P2WSH-P2SH"
       : walletConfig.addressType;
 
-    const interaction = ConfigAdapter({
+    const interactionAdapter = ConfigAdapter({
       KEYSTORE: COLDCARD,
       jsonConfig: walletConfig,
     });
-    const body = interaction.adapt();
+    const body = interactionAdapter.adapt();
     const filename = `wc-${walletConfig.name}.txt`;
     downloadFile(body, filename);
   };
+  const handlePSBTDownloadClick = () => {
+    const body = interaction.request().toBase64();
```

Comment:
Should this be `interactionAdapter`? Not sure if I'm missing a declaration elsewhere. 

```suggestion
    const body = interactionAdapter.request().toBase64();
```

Reply:
This is a great point and something that confused me a lot in this file and a few others. The problem here is a naming conflict between a prop and a variable. Class components allow certain props, methods, or variables to share names, but functional components do not allow this.

In this instance `ColdCardSigner` has a prop called `interaction.` Inside `reshapeConfig`, a new variable was defined with the same name as the prop: “interaction,” but was only available locally to “reshapeConfig”. When `ColdCardSigner` was updated to a functional component, `interaction` was defined as a globally-scoped prop. I renamed `reshapeConfig’s` locally-scoped variable `interactionAdapter` to avoid this naming conflict.

Line 45 is scoped inside of `handlePSBTDownloadClick,` which would not have access to `reshapeConfig’s` variables. Thus on line 45, `body = interaction.request().toBase64()` is utilizing `interaction` the prop, not the locally-scoped variable to which it doesn’t have access.

Reply:
Ah ha! Yeah I thought there might be something like that. Hard to see when looking at diffs especially with a refactor like this as you say. Thanks for the clarification!